preview-src/comment.tsx

```diff
@@ -0,0 +1,159 @@
+import * as React from 'react';
+import { useContext, useState, useEffect, useRef } from 'react';
+
+import Markdown from './markdown';
+import { Spaced, nbsp } from './space';
+import { Avatar, AuthorLink } from './user';
+import Timestamp from './timestamp';
+import { Comment } from '../src/common/comment';
+import { PullRequest } from './cache';
+import PullRequestContext from './context';
+import { editIcon, deleteIcon } from './icon';
+
+export type Props = Partial<Comment & PullRequest>;
+
+export function CommentView(comment: Props) {
+	const { id, pullRequestReviewId, canEdit, canDelete, user, author, htmlUrl, createdAt, bodyHTML, body, } = comment;
+	const [ bodyMd, setBodyMd ] = useState(body);
+	const { deleteComment, editComment, pr } = useContext(PullRequestContext);
+	const currentDraft = pr.pendingCommentDrafts && pr.pendingCommentDrafts[id];
+	const [inEditMode, setEditMode] = useState(!!currentDraft);
+	const [showActionBar, setShowActionBar] = useState(false);
+
+	useEffect(() => {
+		if (body !== bodyMd) {
+			setBodyMd(body);
+		}
+	}, [body]);
+
+	if (inEditMode) {
```

Comment:
I think we need more logic about where the edit comment component gets rendered.

For comments in a review, it should be inside the comment body so that the header is still visible when editing and there's some padding around it, right now it looks like:
![Screen Shot 2019-05-06 at 9 09 02 AM](https://user-images.githubusercontent.com/3672607/57238593-9c775580-6fde-11e9-8d5a-8b27188e16d8.png)

For review comments, the entire review is being replaced:
![review_edit](https://user-images.githubusercontent.com/3672607/57238676-d34d6b80-6fde-11e9-8cbc-9a0357751f15.gif)

I think description is the special case where replacing all of it with the editing component makes sense

Reply:
Fixed.